### DISCUSSION TITLE
adding CURLOPT_PINNEDPUBLICKEY_BLOB

### DISCUSSION THREAD

**Discussion by gvollant** [curl/curl#6822] [Ideas]
Hello,
near all cert file option have now the blob option
https://github.com/curl/curl/pull/6662 add CURLOPT_CAINFO_BLOB

but pehaps we must add CURLOPT_PINNEDPUBLICKEY  ?

**Comment by jay** [comment]
[CURLOPT_PINNEDPUBLICKEY](https://curl.se/libcurl/c/CURLOPT_PINNEDPUBLICKEY.html):
> The string can also be any number of base64 encoded sha256 hashes preceded by "sha256//" and separated by ";"

Seems sufficient to me. There is a maintenance cost with every feature that's added.

**Reply by gvollant** [reply]
I agree
